services/web-server/src/utils/regenerateSession.js

```diff
@@ -0,0 +1,13 @@
+// Regenerate a sessionId given a request
+// https://stackoverflow.com/a/26394156/6150432
```

Comment:
Even reading this stackoverflow, I can't quite figure out what this is for.. help?

Reply:
Passport adds a `passport` key to `req.session`. When `req.session.regenerate` runs,  `req.session.passport` becomes undefined since `req.session` gets recreated. The callback makes the necessary changes to make passport aware of the current user.

I'll add a comment.